sonar-scanner-engine/src/main/java/org/sonar/scanner/phases/SensorsExecutor.java

```diff
@@ -28,21 +28,34 @@
 import org.sonar.scanner.bootstrap.ScannerExtensionDictionnary;
 import org.sonar.scanner.events.EventBus;
 import java.util.Collection;
+import org.sonar.scanner.sensor.SensorScope;
 
 @ScannerSide
 public class SensorsExecutor {
   private final EventBus eventBus;
   private final DefaultInputModule module;
   private final ScannerExtensionDictionnary selector;
+  private final SensorScope scope;
 
-  public SensorsExecutor(ScannerExtensionDictionnary selector, DefaultInputModule module, EventBus eventBus) {
+  public SensorsExecutor(ScannerExtensionDictionnary selector, DefaultInputModule module, EventBus eventBus, SensorScope scope) {
     this.selector = selector;
     this.eventBus = eventBus;
     this.module = module;
+    this.scope = scope;
   }
 
   public void execute(SensorContext context) {
-    Collection<Sensor> sensors = selector.select(Sensor.class, module, true, null);
+    execute(context, false);
+    if (isRoot(module)) {
+      boolean orig = scope.isGlobal();
+      scope.setGlobal(true);
+      execute(context, true);
+      scope.setGlobal(orig);
+    }
+  }
+
+  private void execute(SensorContext context, boolean global) {
+    Collection<Sensor> sensors = selector.selectSensor(global, module, true, null);
     eventBus.fireEvent(new SensorsPhaseEvent(Lists.newArrayList(sensors), true));
```

Comment:
This event should probably be raised only once for the whole module. If not I'm afraid of possible side effects on profiling